packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -78,7 +78,7 @@ class FlatTermSelector extends Component {
 
 	componentDidMount() {
 		if ( ! isEmpty( this.props.terms ) ) {
-			this.setState( { loading: false } );
+			this.setState( { loading: true } );
```

Comment:
Hi @tfrommen thank you for your PR. Would it work if we set the loading state true when ! isEmpty( this.props.terms ) on the constructor instead of on componentDidMount? The reason being we would avoid an unnecessary rerender that happens when setting state via componentDidMount.

Reply:
Hi @jorgefilipecosta - hm, sure, I could do that. And I understand your reasoning.

On the other hand, it somehow feels a bit strange to split the actual loading logic (including the subsequent `loading: false`) from the part where we set the loading state to be `true` (in the constructor). 🤔 🤷‍♂️

Reply:
Hi @tfrommen,
On the constructor, we are initializing the state to false `loading: false,`. I don't think it is wrong to initialize to true or false depending on if this.props.terms is empty or not. If terms exist we know that the component will start the loading process.
Basically, we would set the loading on the constructor and unset the loading when the request is finished, I agree the code would look better if this logical parts were together but given the performance reason I think the separation is ok.

Reply:
@jorgefilipecosta updated as per your feedback.